### ISSUE TITLE
Trouble installing glmmTMB package in R in conda environment

### ISSUE THREAD

**Issue by jamesyitan** [dyxstat/HiCBin#7]
Hi,

After activating the conda environment, opening up R and then running the "install.packages("glmmTMB", type="source")" command, it seems that R has trouble finding the C compiler within the conda environment for all of the dependencies for glmmTMB in R. Here is the error message for each package:

```
* installing *source* package ‘utf8’ ...
** package ‘utf8’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
x86_64-conda-linux-gnu-cc -I"/home/jamestan/.conda/envs/HiCBin_env/lib/R/include" -DNDEBUG   -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/jamestan/.conda/envs/HiCBin_env/include -I/home/jamestan/.conda/envs/HiCBin_env/include -Wl,-rpath-link,/home/jamestan/.conda/envs/HiCBin_env/lib  -Iutf8lite/src -fpic  -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/jamestan/.conda/envs/HiCBin_env/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/r-base-split_1611775910217/work=/usr/local/src/conda/r-base-4.0.3 -fdebug-prefix-map=/home/jamestan/.conda/envs/HiCBin_env=/usr/local/src/conda-prefix  -c as_utf8.c -o as_utf8.o
x86_64-conda-linux-gnu-cc: fatal error: cannot execute 'cc1': execvp: No such file or directory
compilation terminated.
make: *** [/home/jamestan/.conda/envs/HiCBin_env/lib/R/etc/Makeconf:174: as_utf8.o] Error 1
ERROR: compilation failed for package ‘utf8’
```

I've tried updating to a more recent version of conda (23.1.0) but that doesn't seem to fix it either. Wondering if you had any thoughts on how to move forward.

Thanks!

**Comment by dyxstat** [comment]
Hi, thanks for trying our software.

I think this problem is derived from your gcc calling. Since I did not meet such a problem before, I cannot provide you with specific suggestions. But the following link may help you:

https://stackoverflow.com/questions/11912878/gcc-error-gcc-error-trying-to-exec-cc1-execvp-no-such-file-or-directory

**Comment by jamesyitan** [comment]
Thanks for the response! I looked at a lot of similar issues online but none of them seemed to work. I even tried running HiCBin on my local computer with the conda environment (macOS) but I came upon similar c++ dependency issues.

I decided to try the alternative method of getting glmmTMB onto R with "conda install -c conda-forge r-glmmtmb". When I tried running HiCBin with the test data, I didn't get the error that you said would come up ('Error in .Call("FreeADFunObject", ptr, PACKAGE = DLL) : "FreeADFunObject" not available for .Call() for package "glmmTMB"'). I only received an error saying that glmmTMB was built with a slightly newer R version: "WARNING... R[write to console]: package ‘glmmTMB’ was built under R version 4.0.5". Otherwise it seems the test dataset runs and no other errors occur during normalization.

Does that sound fine? I can also compare the generated test dataset output files with expected output files to make sure the normalization ran fine.
